docs/notes.md

```diff
@@ -603,4 +603,242 @@ sends a pair \\((L\_j,R\_j)\\) of points at each step \\(j = k\dots1\\). An
+\end{aligned}
+\\]
+
+Proving vectors of statements with a single statement
+-----------------------------------------------------
+
+We want to combine the above three statements into a single statement for party `j`. We will also introduce challenge values \\(y_j\\) and \\(z_j\\) that are unique to each party `j`, and use them to help combine the statements. Since these challenge values are independent for each party, we can later merge the per-party combined statements into one statement for all `m` parties.
+
+First, we will combine each of the two vector-statements into a single statement using the verifier's choice of challenge value \\(y\\) that is shared across all parties, and offset by \\(y_j = y^{j \cdot n}\\) that is unique to each party `j`:
+
+\\[
+\begin{aligned}
+  {\langle {\mathbf{a}}\_{Lj}, {\mathbf{2}}^{n} \rangle} &= v \\\\
+  {\langle {\mathbf{a}}\_{Lj} - {\mathbf{1}} - {\mathbf{a}}\_{Rj}, {\mathbf{y}}^{n} \cdot y_j \rangle} &= 0 \\\\
+  {\langle {\mathbf{a}}\_{Lj}, {\mathbf{a}}\_{Rj} \circ {\mathbf{y}}^{n} \cdot y_j \rangle} &= 0
+\end{aligned}
+\\]
+
+The three resulting statements can then be combined in the same way,
+using the verifier’s choice of challenge value \\(z\\) that is shared across all parties, and offset by \\(z_j = z^j\\) that is unique to each party `j`:
+\\[
+\begin{aligned}
+z^{2} z_j \cdot v 
+&= 
+   z^{2} z_j \cdot {\langle {\mathbf{a}}\_{Lj}, {\mathbf{2}}^{n} \rangle} \\\\
+     &+ z \cdot {\langle {\mathbf{a}}\_{Lj} - {\mathbf{1}} - {\mathbf{a}}\_{Rj}, {\mathbf{y}}^{n} \cdot y_j \rangle} \\\\
+         &+   {\langle {\mathbf{a}}\_{Lj}, {\mathbf{a}}\_{Rj} \circ {\mathbf{y}}^{n} \cdot y_j \rangle} 
+\end{aligned}
+\\]
+
+Combining inner products
+------------------------
+
+We combine the terms in the last statement into a single inner product, using the same technique as in the single-value range proof. We will not reproduce the math here, only the end result:
+
+\\[
+\begin{aligned}
+ \delta_j(y,z) &= (z - z^{2}) \cdot {\langle {\mathbf{1}}, {\mathbf{y}}^{n} \cdot y_j \rangle} - z^{3} z_j \cdot {\langle {\mathbf{1}}, {\mathbf{2}}^{n} \rangle}\\\\
+ z^{2}z_j \cdot v + \delta_j(y,z) &= {\langle {\mathbf{a}}\_{Lj} - z {\mathbf{1}}, {\mathbf{y}}^{n} \cdot y_j \circ ({\mathbf{a}}\_{Rj} + z {\mathbf{1}}) + z^{2} z_j \cdot {\mathbf{2}}^{n} \rangle}
+\end{aligned} 
+\\]
+
+Blinding the inner product
+--------------------------
+
+The prover chooses vectors of blinding factors
+\\[
+{\mathbf{s}}\_{Lj}, {\mathbf{s}}\_{Rj} \\;{\xleftarrow{\\$}}\\; {\mathbb Z\_p}^{n}
+\\]
+and uses them to construct blinded vector polynomials:
+\\[
+\begin{aligned}
+  {\mathbf{l}}\_j(x) &= ({\mathbf{a}}\_{Lj} + {\mathbf{s}}\_{Lj} x) - z {\mathbf{1}} & \in {\mathbb Z\_p}[x]^{n}  \\\\
+  {\mathbf{r}}\_j(x) &= {\mathbf{y}}^{n} \cdot y_j \circ \left( ({\mathbf{a}}\_{Rj} + {\mathbf{s}}\_{Rj} x\right)  + z {\mathbf{1}}) + z^{2} z_j {\mathbf{2}}^{n} &\in {\mathbb Z\_p}[x]^{n} 
```

Comment:
Previously, `y_j = y^j` was introduced as a new variable (I guess the intent is to distinguish its role as a per-party offset, rather than its construction as `y^j` ??). In that case, does it make sense to *also* introduce `\mathbf{y}_j` as the per-party view into the larger exponent vector, for the same reasons?

Reply:
oo that's a great idea!

Reply:
and it was introduced as `y_j = y^(j*n)` so readers wouldn't have to worry about the exponent, just that it's the `j`th party's share of `y`. But your suggestion is even better / clearer than that!